pt_miniscreen/core/app.py

```diff
@@ -0,0 +1,26 @@
+import logging
+
+from PIL import Image
+
+logger = logging.getLogger(__name__)
+
+
+class App:
+    def __init__(self, miniscreen=None, Root=None):
+        assert miniscreen is not None
+        assert Root is not None
+        self.miniscreen = miniscreen
+        self.Root = Root
+
+    def start(self):
+        self.root = self.Root(on_rerender=self.display)
+        self.display()
+
+    def stop(self):
+        self.root._cleanup()
+
+    def display(self):
+        logger.debug("Update display")
+        self.miniscreen.device.display(
```

Comment:
why do this vs miniscreen.display_image?

Reply:
I wasn't aware of that method! Also this is what the previous implementation used.

Reply:
right fair enough. display_image is the main official api but has a bunch of 'saftey' features like image resizing, stopping currently running gif animations, and fps regulation

using display_image shouldn't cause issues and whilst it does add a bunch of code that's unnecessary for the pt-miniscreen case, sdk users could run into these cases. eg if they are using the fps throttling feature and running an animation, then start one of these apps, they might expect it to stop the animation and respect the fps they set, which display_image will do

Reply:
Oh interesting! Yeah if there are features of the miniscreen class that it enables at a high level then that should be what we use for sure. I'll try to swap over to it and try out some of the features it enables to make sure it doesn't cause any unexpected issues.

Reply:
I looked into using `display_image` but found an issue with using it right now. I've added a comment with an explanation. It should be pretty easy to move to it once the opencv import issue is resolved.